ubiquity/dell-bootstrap.py

```diff
@@ -1360,6 +1348,14 @@ def install(self, target, progress, *args, **kwargs):
 
         #install dell-recovery only if there is an RP
         if rec_part:
+            #hide the recovery partition as default
+            try:
+                recovery = magic.find_factory_partition_stats()
```

Comment:
I think you might need to change find_factory_partition_stats() because it might be mistakingly finding a Windows partition.  It needs to know what to do in dual boot scenarios (probably needs to look for the label for dual boot recovery before other labels)